packages/strapi-connector-mongoose/lib/queries.js

```diff
@@ -4,13 +4,16 @@
  */
 
 const _ = require('lodash');
+const mongoose = require('mongoose');
 const { convertRestQueryParams, buildQuery } = require('strapi-utils');
 
 const { findComponentByGlobalId } = require('./utils/helpers');
 
 const hasPK = (obj, model) => _.has(obj, model.primaryKey) || _.has(obj, 'id');
 const getPK = (obj, model) => (_.has(obj, model.primaryKey) ? obj[model.primaryKey] : obj.id);
 
+const isValidId = id => mongoose.Types.ObjectId.isValid(id);
```

Comment:
using this method is actually not working well you should be using  the utils `isMongoId` availaible in this package utils

Reply:
Done